### PR TITLE
Add prefer-to-be-null rule

### PR DESCRIPTION
**Pull request by xfumihiro**

### CODE REVIEW

File: rules/prefer_to_be_null.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        node.arguments.length == 1 &&
+        node.parent &&
+        node.parent.type === 'MemberExpression' &&
+        node.parent.parent
+      ) {
+        const parentProperty = node.parent.property;
+        const propertyName = parentProperty.name;
+        const argument = node.parent.parent.arguments[0];
+
+        if (propertyName === 'toBe' && argument.value === null) {
```
Comment (SimenB) [inline rules/prefer_to_be_null.js]: check `toEqual` as well

File: rules/__tests__/prefer_to_be_null.test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+const RuleTester = require('eslint').RuleTester;
+const rules = require('../../').rules;
+
+const ruleTester = new RuleTester();
+
+ruleTester.run('prefer_to_be_null', rules['prefer-to-be-null'], {
+  valid: ['expect(null).toBeNull();'],
+
+  invalid: [
```
Comment (SimenB) [inline rules/__tests__/prefer_to_be_null.test.js]: Could you add a test for `toBeEqual` here (and the others)?

File: README.md [hunk 3]
```diff
@@ -32,6 +32,7 @@ Then configure the rules you want to use under the rules section.
     "jest/no-disabled-tests": "warn",
     "jest/no-focused-tests": "error",
     "jest/no-identical-title": "error",
+    "jest/prefer-to-be-null": "warn",
```
Comment (SimenB) [inline README.md]: IMO this should be off by default, otherwise it's a breaking change

**Comment by SimenB** [comment]
I never got an email that you pushed, sorry about being slow here!

**Comment by xfumihiro** [comment]
No worries!

File: README.md (hunk 3 above)
Comment (xfumihiro) [reply README.md]: Why so?
Same for all others?

Comment (SimenB) [reply README.md]: If I use `toBe(null)` somewhere in my code, and have activated the recommended rules, my build is potentially broken in a feature release (or we can do a major, but I don't think that's necessary).

other open PRs yes

Comment (thymikee) [reply README.md]: It's a "warn" level, so by default it should not fail the lint command.

Comment (xfumihiro) [reply README.md]: Just removed all warnings. 😓 
Should redo?

Comment (SimenB) [reply README.md]: At work, we have `--no-warnings`

Comment (xfumihiro) [reply README.md]: How about moving forward on this at the moment?

Comment (SimenB) [reply README.md]: Oh, sorry! yeah, let's just roll with it

**Pull request merged**